### ISSUE TITLE
Consistency/accuracy of gps locations

### ISSUE THREAD

**Issue by LarsBocons** [transistorsoft/cordova-background-geolocation-lt#607]
I've been holding this back for a while, since I have not yet come up with something precise to pinpoint, but we seem to have problems consistantly receiving positions. It's all a bit fuzzy and inconsistant, so all we can do is to try and reach out blindfolded:

## Your Environment
* Plugin version: Latest
* Platform: iOS and Android
* OS version: Android 6 and iOS 11.2.5
* Device manufacturer / model: Huawei 7 Honor, iPad air, iPhone X
* Cordova version 7.1.0 (PhoneGap cli-7.1.0)
* Cordova platform version ?
* Plugin config
```
var config = {
            desiredAccuracy: 0,
            distanceFilter: 0,
            disableElasticity: true,
            stationaryRadius: 1,
            locationAuthorizationRequest: "Always",
            locationAuthorizationAlert: {
                titleWhenNotEnabled: "Lokalitetstjenester er ikke aktiveret",
                titleWhenOff: "Lokalitetstjenester er slået fra",
                instructions: "Denne app kræver at 'Altid' er valgt i lokalitetstjenester",
                cancelButton: "Fortryd",
                settingsButton: "Indtillinger"
            },
            locationUpdateInterval: 900,
            fastestLocationUpdateInterval: 900,
            activityRecognitionInterval: 500,
            heartbeatInterval: 30,
            preventSuspend: true, //Enable this to prevent iOS from suspending your app in the background while in the stationary state. Must be used in conjunction with a #heartbeatInterval.
            foregroundService: true, //Android: Running as a foreground-service makes the tracking-service much more inmmune to OS killing it due to memory/battery pressure
            stopTimeout: 60, //The number of minutes to wait before turning off location- services after the ActivityRecognition System (ARS) detects the device is STILL
            debug: true //enable this hear sounds for background-geolocation life-cycle.
        }
```

## Expected Behavior
Excpect to receive at least one location per second (at least when moving)

## Actual Behavior

What we have experienced in out test, is that the positions seems to take a long time to reach our required accuracy (30 meters). Also there have been gaps of 1-10 minutes between positions (debug sound verifies this) while moving. Our app is running in front and use power management to prevent the screen from turning off. 

We have also experienced what seems to be a 10 minute suspend (even though it looked like the app was running), a javacript timer increments a counter every 30 seconds, but took 10 minutes to increment. 
Is there any way the component could affect the javascript in this way?

Problem seems to be mostly affecting iOS, but even with Android it seems like it's harder to get a position with acceptable accuracy.
Could there be something in the component that turns down the throttle? We just want to turn it up to full, with no regard to battery consumption.

We have so far solved the slow-accuracy problem by supplementing with HTM5 navigator reception, but we would rather just use one source of gps location. Do you se any problem with this approach by the way?

**Comment by christocracy** [comment]
>Is there any way the component could affect the javascript in this way?

No.

>Could there be something in the component that turns down the throttle?

No.  Only the OS.

If the device is *moving*, the plugin has requested locations from the device according to your configuration. 

This is 7 days of uninterrupted tracking in my home town during xmas:
![](https://dl.dropboxusercontent.com/s/kb1dsgl67ukx3jx/Screenshot%202018-02-19%2008.56.55.png?dl=1).  This app was not touched during this time.  It never failed to miss a beat.  If you're experiencing issues, it's either the OS or the network environment.

**Comment by LarsBocons** [comment]
Ok, as I said it's all a bit random, so hard to pinpoint anything. 

We have also experienced a few "MainThread" & "WebThread" app crashes after updating to iOS 11, which seem unrelated to your component.

In theory a hybrid app should never crash, right? I mean, no matter how much you mess up the javascript, it should just do nothing, or display an error ( if you told it too) ?

**Comment by LarsBocons** [comment]
With "in theory" I mean assuming that no plugins are used, no pure javascript should be able to crash an app.

During gps reception there's no interaction with other plugins. localStorage is used to temporary hold on to locations, but that is pure javascript as far as I know.
The error we got a few times now, is exactly the same as this one: [https://github.com/NativeScript/NativeScript/issues/5019](url)
...except it is caused by the WebView thread.

**Comment by christocracy** [comment]
I don't know what to tell you.  It's not related to this plugin if you don't have a stacktrace referencing the plugin.

**Comment by LarsBocons** [comment]
Sorry, you must have missed the "which seem unrelated to your component" comment above. I'm not blaming your plugin for the crashes, just posting info in hope that maybe someone will put 2 & 2 together. 

The code has worked for years without ever crashing, but we had to do something about the new iOS 11 security additions. So the main suspect is iOS 11. 

I have now updated to latest vesions of all included cordova plugins, and will see if that changes anything.
